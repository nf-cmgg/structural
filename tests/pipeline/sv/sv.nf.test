nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"

    tag "pipeline"
    tag "pipeline_sv"

    options "-stub"

    test("one caller") {

        when {
            params {
                outdir = "${outputDir}"
                callers = "delly"
                max_cpus   = 2
                max_memory = '6.GB'
                max_time   = '6.h'
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.stdout,

                    path("$outputDir").list().findAll{
                        !it.toString().endsWith("pipeline_info")
                    },
                    path("$outputDir").list().collect{
                        it.toString().split("/")[-1]
                    }
                ).match("delly")}
            )
        }

    }

    test("all callers") {

        when {
            params {
                outdir = "${outputDir}"
                callers = "delly,manta,smoove,gridss"
                max_cpus   = 2
                max_memory = '6.GB'
                max_time   = '6.h'
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.stdout,

                    path("$outputDir").list().findAll{
                        !it.toString().endsWith("pipeline_info")
                    },
                    path("$outputDir").list().collect{
                        it.toString().split("/")[-1]
                    }
                ).match("all")}
            )
        }

    }

}
