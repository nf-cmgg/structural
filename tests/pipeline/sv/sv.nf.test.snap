{
    "all": {
        "content": [
            [
                
            ],
            [
                [
                    "PosCon1.sv.vcf.gz",
                    "PosCon1.sv.vcf.gz.tbi"
                ],
                [
                    "PosCon2.sv.vcf.gz",
                    "PosCon2.sv.vcf.gz.tbi"
                ],
                [
                    "PosCon3.sv.vcf.gz",
                    "PosCon3.sv.vcf.gz.tbi"
                ],
                [
                    "PosCon4.sv.vcf.gz",
                    "PosCon4.sv.vcf.gz.tbi"
                ],
                [
                    "multiqc_data",
                    "multiqc_plots",
                    "multiqc_report.html"
                ]
            ],
            [
                "PosCon1",
                "PosCon2",
                "PosCon3",
                "PosCon4",
                "multiqc",
                "pipeline_info"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.02.0"
        },
        "timestamp": "2024-03-13T13:07:19.225114007"
    },
    "delly": {
        "content": [
            [
                
            ],
            [
                [
                    "PosCon1.sv.vcf.gz",
                    "PosCon1.sv.vcf.gz.tbi"
                ],
                [
                    "PosCon2.sv.vcf.gz",
                    "PosCon2.sv.vcf.gz.tbi"
                ],
                [
                    "PosCon3.sv.vcf.gz",
                    "PosCon3.sv.vcf.gz.tbi"
                ],
                [
                    "PosCon4.sv.vcf.gz",
                    "PosCon4.sv.vcf.gz.tbi"
                ],
                [
                    "multiqc_data",
                    "multiqc_plots",
                    "multiqc_report.html"
                ]
            ],
            [
                "PosCon1",
                "PosCon2",
                "PosCon3",
                "PosCon4",
                "multiqc",
                "pipeline_info"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.02.0"
        },
        "timestamp": "2024-03-13T13:04:12.383695256"
    }
}